### PR TITLE
Add config to docker image

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <ploffay@redhat.com>

**Description:** <Describe what has changed. 

Add `examples/otel-local-config.yaml` into docker image to make it easier for users to get a default config.

@tigrannajaryan could you please have a look?

**Link to tracking Issue:** <Issue number if applicable>

Updates #886 

Follow up PR should add config into other distributions and add comments into the config.

**Testing:** < Describe what testing was performed and which tests were added.>

Tested docker build locally.

**Documentation:** < Describe the documentation added.>

Once this is merged I will update Docker instructions https://opentelemetry.io/docs/collector/about/ to use 
```
docker create --name otelcol-config otelcol
docker cp otelcol-config:/config.yaml . && docker rm otelcol-config
// Edit the config.yaml
docker run  -v "${PWD}/config.yaml":/config.yaml  otelcol --config /config.yaml
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -189,6 +189,7 @@ delete-tag:
 
 .PHONY: docker-otelcol
 docker-otelcol:
+	cp examples/otel-local-config.yaml cmd/otelcol
```
Comment (pavolloffay) [inline Makefile:192]: This is not nice but seems better than setting the docker context to the root of the project.

**Comment by tigrannajaryan** [comment]
@pavolloffay this merely places the example config file in the docker image so that it is available for the user as a starting point for their custom config file, right?

**Comment by pavolloffay** [comment]
Correct see the example commands in the first comment on how the config can be edited (first copy to host then edit). Or users can run it directly with log exporters for quick demo purposes.

**Review by tigrannajaryan — APPROVED** [review]
LGTM. Will wait a bit before mering to see @owais has any comments.

**Comment by owais** [comment]
Contrib docker image uses `/etc/otel/config.yaml` as the default path for config file. So running the following command works:

```
docker -v $PWD/local/:/etc/otel otelcontribcol
```
assuming there is a config file at `./local/config.yaml`.

Can we make the following changes?

1. Change docker entrypoint and command to:

```dockerfile
ENTRYPOINT ["/otelcol"]
CMD ["--config", "/etc/otel/config.yaml"]
```

2. Copy the default config file to `/etc/otel/config.yaml` in the image. 


This would ship a default config with the collector while being consistent with the contrib image. We should probably have a default config file for contrib image as well once this is merged.

**Comment by owais** [comment]
I'm not sure about shipping the example config as the default one. I think the default config should be more or less usable out of the box and needs to be vetted more carefully compared to an example config file. Also an example config might contains a lot of stuff than we want to enable by defaults. So probably a separate dedicated default config file would be better (even if it's a 1:1 copy of example config today).

**Comment by pavolloffay** [comment]
@owais I have made the proposed changes to the dockerfile.

> I think the default config should be more or less usable out of the box and needs to be vetted more carefully compared to an example config file. Also an example config might contains a lot of stuff than we want to enable by defaults. So probably a separate dedicated default config file would be better (even if it's a 1:1 copy of example config today).

The default and example configs should be both ready to run out of the box. At the moment I don't see any difference between between example and default config. Both should use sane components and document what is being used for users to understand how to change the config. 

I have copied the example config to the otel dir for now.

File: cmd/otelcol/config.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+exporters:
+  logging:
+    logLevel: debug
+
+service:
+
+  pipelines:
+
+    traces:
+      receivers: [opencensus, jaeger, zipkin]
```
Comment (owais) [inline cmd/otelcol/config.yaml:43]: I think we should enable OTLP by default. I think it should be the default or recommended receiver for all data types.

**Review by owais — APPROVED** [review]
LGTM but let's enable OTLP in the default config file.

**Comment by pavolloffay** [comment]
@owais I will submit a follow-up PR to enable OTLP in the example and default configs.

**Pull request merged**